packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -57,7 +58,17 @@ export interface PluginPackage {
     extensionPack?: string[];
 }
 export namespace PluginPackage {
+    function isRunningInElectron(): boolean {
+        const isElectronMain = typeof process !== 'undefined' && process.versions.electron !== undefined;
+        const isElectronRenderer = typeof navigator !== 'undefined' && navigator.userAgent.toLowerCase().includes('electron');
+        const isVsCodeExtensionInElectron = typeof process !== 'undefined' && process.env.ELECTRON_RUN_AS_NODE !== undefined;
+        return (isElectronMain || isElectronRenderer || isVsCodeExtensionInElectron);
+    }
+
     export function toPluginUrl(pck: PluginPackage, relativePath: string): string {
+        if (isRunningInElectron()) {
```

Comment:
Would it be enough to simply use?
```ts
environment.electron.is()
```

I don't think there's a need to have `isRunningInElectron()`.

Reply:
> Would it be enough to simply use?

Reply:
path.join works with relative paths like `../../to/path`

Reply:
True, I thought this is a Theia `Path`. 👍 The string concatenation part is still not correct, for instance you ignore the fact when there is whitespace on the path, your URI will be `file:///some/path to a whitespace/folder`. I am not sure we can accept it as is.

Reply:
Whitespaces and everything else except for slashes are now escaped with `encodeURIComponent`. In the previous return statement below `relativePath` is also simply concatenated so it should be fine in our case as well.